bpfd/src/bpf.rs

```diff
@@ -48,13 +48,138 @@ const MAPS_MODE: u32 = 0o0660;
 
 pub(crate) struct BpfManager {
     config: Config,
-    dispatchers: HashMap<DispatcherId, Dispatcher>,
-    programs: HashMap<Uuid, Program>,
+    dispatchers: BpfDispatchers,
+    programs: BpfPrograms,
     maps: HashMap<Uuid, BpfMap>,
     commands: Receiver<Command>,
     image_manager: Sender<ImageManagerCommand>,
 }
 
+pub(crate) struct BpfPrograms {
+    programs: HashMap<Uuid, Program>,
+}
+
+impl BpfPrograms {
+    fn new() -> Self {
+        BpfPrograms {
+            programs: HashMap::new(),
+        }
+    }
+
+    fn insert(&mut self, id: Uuid, prog: Program) -> Option<Program> {
+        self.programs.insert(id, prog)
+    }
+
+    fn remove(&mut self, id: &Uuid) -> Option<Program> {
+        self.programs.remove(id)
+    }
+
+    fn get_mut(&mut self, id: &Uuid) -> Option<&mut Program> {
+        self.programs.get_mut(id)
+    }
+
+    fn get(&self, id: &Uuid) -> Option<&Program> {
+        self.programs.get(id)
+    }
+
+    fn contains_key(&self, id: &Uuid) -> bool {
+        self.programs.contains_key(id)
+    }
+
+    fn next_avail_id(&self, program: &Program) -> usize {
```

Comment:
1. is easy enough... but how do we get the number of attached extensions since we don't actually track the extensions in the dispatcher currently.

I added a new field to the Dispatcher's called `num_extensions` but PTAL